File: atcoder/past201912_h.cpp
// https://atcoder.jp/contests/past201912-open/tasks/past201912_h
#include<iostream>
// #include<algorithm>
// #include<string>
// #include<numeric>
// #include<vector>
// #include<map>
// #include<tuple>
// #include<set>
// #include<queue>
// #include<deque>
// #include<regex>
// #include<bitset>
// #include<iomanip>
// #include<complex>
// #include<stack>
// #include<functional>

#include<bits/stdc++.h>
using namespace std;
#define rep(i,n) for (int i = 0; i < (n); ++i)
#define all(v) (v).begin(),(v).end()
#define chmin(x,y) x = min(x,y)
#define chmax(x,y) x = max(x,y)
typedef pair<int, int> P;
typedef long long ll;
const int INF = 1001001001;
const ll LINF = 1001002003004005006ll;
const ll MOD = 1e9+7;
ll gcd(ll a, ll b) { return b?gcd(b,a%b):a;}
ll lcm(ll a, ll b) { return a/gcd(a,b)*b;}

int main() {
	int n;
	cin >> n;
    vector<ll> c(n);
    ll minall = INF, minodd = INF;
    rep(i,n) {
        cin >> c[i];
        minall = min(minall, c[i]);
        if(i%2==0) minodd = min(minodd, c[i]);
    }
    ll sellall = 0, sellodd = 0;
    vector<ll> sell(n,0);
    int q, type;
    ll x, a;
    cin >> q;
    rep(i,q) {
        cin >> type;
        if(type==1) {
            cin >> x >> a;
            if(c[x-1]-sell[x-1]-sellall-sellodd*(x%2) >= a) {
                sell[x-1] += a;
                minall = min(minall, c[x-1]-sell[x-1]-sellall-sellodd*(x%2));
                if(x%2) minodd = min(minodd, c[x-1]-sell[x-1]-sellall-sellodd);
            }
        } else if (type==2) {
            cin >> a;
            if(minodd >= a) {
                sellodd += a;
                minodd -= a;
                minall = min(minall, minodd);
            }
        } else {
            cin >> a;
            if(minall >= a) {
                sellall += a;
                minall -= a;
                minodd -= a;
            }
        }
    }
    ll ans = sellall * n + sellodd * ((n+1)/2);
    rep(i,n) ans += sell[i];
	cout << ans << endl;
	return 0;
}

File: atcoder/past201912_h.py
# https://atcoder.jp/contests/past201912/tasks/past201912_h

N = int(input())
C = list(map(int, input().split()))
Q = int(input())
sm, mineven, minodd = 0, 10**9, 10**9
for i in range(N):
    sm += C[i]
    if i%2:
        mineven = min(mineven, C[i])
    else:
        minodd = min(minodd, C[i])

minuseven, minusodd = 0, 0
for i in range(Q):
    S = list(input().split())
    if S[0] == '1':
        x = int(S[1]) - 1
        a = int(S[2])
        if x%2:
            minus = minuseven
        else:
            minus = minusodd
        if C[x] - minus >= a:
            C[x] -= a
            if x%2:
                mineven = min(mineven, C[x])
            else:
                minodd = min(minodd, C[x])
    elif S[0] == '2':
        a = int(S[1])
        if minodd - minusodd >= a:
            minusodd += a
    else:
        a = int(S[1])
        if min(mineven - minuseven, minodd - minusodd) >= a:
            minuseven += a
            minusodd += a
    # print(i, C, minuseven, minusodd)

# print(sm, C, minuseven, minusodd)
print(sm - sum(C) + minuseven*(N//2) + minusodd*((N+1)//2))
